lib/vagrant/util/platform.rb

```diff
@@ -33,6 +33,22 @@ def cygwin?
           end
         end
 
+        def wsl?
+          return @_wsl if defined?(@_wsl)
+          @_wsl = -> {
+            platform = File.read('/proc/sys/kernel/osrelease')
+            return platform.include?("Microsoft")
+          }.call
+          return @_wsl
```

Comment:
Honest answer: this is the first time I've written Ruby code and I copied it from above :)